src/App.vue

```diff
@@ -26,6 +26,11 @@ export default {
     isSmallDevice() {
       return this.$vuetify.display.smAndDown;
     },
+    navToggle() {
+      const isHomeView = this.$router.currentRoute.value.name === 'Home';
+      const linkColor = isHomeView ? 'text-surface-1' : 'text-surface-2'; 
+      return {isHomeView, linkColor};
+    }
```

Comment:
Let's make `<the-header>` and `<the-footer>` 'dumber' about the color by giving each of them a `bg-color` and `link-color` prop directly so they don't need to individually care about which page we're on, App.vue can do that calculation and then the child components just render whatever color they're given. That way if we update another page, we only have to update the logic for caring about which page is which in one place.
```suggestion
    isDarkBackground() {
      return this.$router.currentRoute.value.name === 'Home';
    },
    navBackground() {
      return this.isDarkBackground ? 'surface-2' : 'surface-1'; 
    },
    navLinkColor() {
      return this.isDarkBackground ? 'text-surface-1' : 'text-surface-2';
    },
```

Reply:
Actually scrap that. I think we will likely want to have light and dark variants of a number of base components including these. So instead, how about we give each of them a prop called 'theme' which could be `'light'` or `'dark'` and they compute the background and navLink colors based on that?

Reply:
yeah that's good idea, how about we store it in vuex store than it will be accessible in every component instead of passing as props.

Reply:
I like the idea of centralizing it, and Vuetify has built in support for swapping entire themes we could use for that purpose, but my concern is that I'm not sure all components will use light/dark variations the same way at the same time, so I'm conflicted. I lean towards props but I'm open to discussion